### PR TITLE
[RFR] Use React Router NavLink in menu

### LINKED ISSUE #1487: Use NavLink instead of Link in MenuItemLink component
To be able to highlight current resource

### PR DESCRIPTION
**Pull request by djhi**
screen

Fixes #1487

### CODE REVIEW

**Comment by djhi** [comment]
We don't have any documentation yet for the `MenuItemLink`. Moreover, we need to document the CSS API for our components which uses `classes`.

**Comment by fzaninotto** [comment]
The use of secondary color is ugly IMHO. Material-ui uses an inverse background, I think it's better

image

File: packages/react-admin/src/mui/layout/MenuItemLink.js [hunk 1]
```diff
@@ -20,6 +24,7 @@ export class MenuItemLink extends Component {
 
     render() {
         const {
+            activeClassName,
```
Comment (fzaninotto) [inline packages/react-admin/src/mui/layout/MenuItemLink.js]: why do you add that prop? Developers can already pass their own `classes`

Comment (djhi) [reply packages/react-admin/src/mui/layout/MenuItemLink.js]: because it's easier ? I'll remove it if you care

Comment (fzaninotto) [reply packages/react-admin/src/mui/layout/MenuItemLink.js]: yes please

**Pull request merged**